components/arm/xarm/xarm.go

```diff
@@ -153,13 +154,16 @@ func (x *xArm) Reconfigure(ctx context.Context, deps resource.Dependencies, conf
 
 	newAddr := net.JoinHostPort(newConf.Host, newConf.parsedPort)
 	if x.conn == nil || x.conn.RemoteAddr().String() != newAddr {
+		// Need a new or replacement connection
 		var d net.Dialer
 		newConn, err := d.DialContext(ctx, "tcp", newAddr)
 		if err != nil {
 			return err
 		}
-		if err := x.conn.Close(); err != nil {
-			x.logger.Warnw("error closing old connection but will continue with reconfiguration", "error", err)
+		if x.conn != nil {
```

Comment:
[q] why is this being added? When will this be true?

Reply:
conn is nil on the initial reconfigure that initializes the arm at robot startup.

conn is not nil when the arm config gets changed.

Reply:
Ok I see, the if statement at L156 got me confused, but yes this makes sense